src/storage/storage_service.ml

```diff
@@ -19,6 +19,13 @@ module Make (Log : Log.Sig.SERVICE) (Repo : REPO) : SERVICE = struct
     | None -> raise (Exception ("File not found with id " ^ id))
     | Some file -> Lwt.return file
 
+  let delete ctx ~id =
+    let* file = find ctx ~id in
+    let blob_id = StoredFile.blob file in
+    let* () = Repo.delete_blob ctx ~id:blob_id in
```

Comment:
@aronerben Please switch the calls and put them into `DbService.atomic`

Reply:
Done as you wish